docs/en/introduction.md

```diff
@@ -0,0 +1,99 @@
+
+```rust
+fn_widget!{
+  @Text {
+    // `margin` is not a field of `Text`,
+    // it is a field of the built-in widget `Margin`,
+    // but it can still be used directly by `Text`.
+    margin: EdgeInset::all(8),
+    text: "Hello world!"
+  }
+}
+```
+
+The above example shows the way of combining built-in widgets. Even if `Text` does not have a `margin` field, you can still use the `Margin::margin` and compose it with `Text` to form a new widget. `Margin` will only be created when a widget uses the `margin` field, otherwise there will be no overhead.
+
+**Digestion of composite widget**: When describing the view of the data, in addition to some basic widgets, most widgets are composed of other widgets. For example, a `Button`, it is composed of `Text`, `Icon` or `BoxDecoration`, etc. widgets, `Button` itself is not a view element, we call this type of widget a composite widget. Composite widgets will be digested during view construction. They are like a function and are called once during view construction to build the final view and create the corresponding update logic, and do not exist in the final view.
+
+**Only state with write sources are real state**: Unlike other declarative frameworks that add fields to widgets to control widget updates. Ribir is non-intrusive. Ribir treats the entire widget as a state to control updates. 
+At the same time, it provides the ability to split the state, so that the local view can directly depend on the modification of part of the data to update (introduced in detail in the subsequent tutorial). Another big difference is that stateful and stateless can be converted to each other. If a state has no write source, it will degenerate into stateless， because no one will update it. For example:
+
+```rust
+fn_widget!{
+  let show_hi = Stateful::new(true);
+  @Text {
+    visible: pipe!($show_hi),
+    text: "Hello world!"
+  }
+}
+```
+
+In the above example, we declared a `Text` and used the `pipe!` macro to directly associate the visibility of `Text` with `show_hi`. But this association will be eliminated when the view is constructed, because `show_hi` always remains unchanged - it has no write source. Therefore, Ribir constructs a simple static view.
+
+### Reliability
+
+Unlike general GUI frameworks that use inheritance and do not have any type constraints except base class inheritance, Ribir builds views based on widget composition and relies on the types between parent and child widgets to constrain whether and how to compose them. You can standardize your own child types, so many errors can be reported at compile time instead of being checked at runtime.
+
+
+## What is the current status of Ribir?
+
+### Stability
+
+The core framework of Ribir is in a basically stable state, and the API and syntax will be iterated with a cautious attitude. Although the widget library already has many available widgets, it is still in a very rough state, and there will be major changes in each version.
```

Comment:
Although the widget library already has many available widgets, it is still in a very rough state, and there will be major changes in each version.
->
Although the widget library already has many available widgets, it is still in a very rough state, and each version will have major changes.